src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3342,13 +3377,20 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     void TerminalPage::_DismissMessage(const InfoBarMessage& message)
     {
-        auto dismissedMessages = ApplicationState::SharedInstance().DismissedMessages();
-        if (!dismissedMessages)
+        const auto applicationState = ApplicationState::SharedInstance();
+        std::vector<InfoBarMessage> messages;
+
+        if (const auto values = applicationState.DismissedMessages())
+        {
+            messages.resize(values.Size());
+            values.GetMany(0, messages);
+        }
+
+        if (std::none_of(messages.begin(), messages.end(), [&](const auto& m) { return m == message; }))
         {
-            dismissedMessages = winrt::single_threaded_vector<InfoBarMessage>();
+            messages.emplace_back(message);
         }
 
-        dismissedMessages.Append(message);
-        ApplicationState::SharedInstance().DismissedMessages(dismissedMessages);
+        applicationState.DismissedMessages(std::move(messages));
```

Comment:
FYI This fixes a pretty ugly programming error. `ApplicationState` is supposed to be a thread-safe class.
Calling `Append` on the returned vector however isn't thread-safe, so I rewrote this.